grakn-graph/src/main/java/ai/grakn/graph/internal/AbstractGraknGraph.java

```diff
@@ -363,10 +363,15 @@ void checkMutationAllowed(){
 
     //----------------------------------------------Concept Functionality-----------------------------------------------
     //------------------------------------ Construction
-    @Nullable
     VertexElement addVertex(Schema.BaseType baseType){
         Vertex vertex = operateOnOpenGraph(() -> getTinkerPopGraph().addVertex(baseType.name()));
-        vertex.property(Schema.VertexProperty.ID.name(), vertex.id().toString());
+        vertex.property(Schema.VertexProperty.ID.name(), Schema.PREFIX_VERTEX + vertex.id().toString());
+        return factory().buildVertexElement(vertex);
+    }
```

Comment:
I tried but the problem is that the one case required the vertex to exist. You basically end up making a lot of really ugly changes just to avoid 2 similar lines.